src/components/Explore/LeftSidebar/InsetMap.tsx

```diff
@@ -0,0 +1,172 @@
+        };
+    }, []);
+
+    const mainMap = mainMapRef.current;
+    const insetMap = insetMapRef.current;
+
+    useEffect(() => {
+        if (insetMap) {
+            runWhenReady(insetMap, () => {
+                BASEMAPS.forEach(({ id }) => {
+                    insetMap.setLayoutProperty(id, 'visibility', id === activeBasemap ? 'visible' : 'none');
+                });
+            });
+        }
+    }, [activeBasemap, insetMap]);
+
+    useEffect(() => {
+        if (mainMap && insetMap) {
+            const syncInsetMap = () => {
+                const bounds = mainMap.getBounds();
+
+                const extentCoords =
+                    bounds.getEast() - bounds.getWest() >= 360
+                        ? [
+                              [-180, bounds.getNorth()],
+                              [180, bounds.getNorth()],
+                              [180, bounds.getSouth()],
+                              [-180, bounds.getSouth()],
+                              [-180, bounds.getNorth()]
+                          ]
+                        : [
+                              bounds.getNorthWest().toArray(),
+                              bounds.getNorthEast().toArray(),
+                              bounds.getSouthEast().toArray(),
+                              bounds.getSouthWest().toArray(),
+                              bounds.getNorthWest().toArray()
+                          ];
+
+                const extent =
+                    turf.union(
+                        turf.union(
+                            turf.polygon([extentCoords.map(([lng, lat]) => [lng - 360, lat])]),
+                            turf.polygon([extentCoords.map(([lng, lat]) => [lng + 360, lat])])
+                        ) ?? turf.polygon([]),
+                        turf.polygon([extentCoords])
+                    ) ?? turf.polygon([]);
+
+                const shadow =
+                    turf.difference(
+                        turf.polygon([
+                            [
+                                [-180, 90],
+                                [180, 90],
+                                [180, -90],
+                                [-180, -90],
+                                [-180, 90]
+                            ]
+                        ]),
+                        extent
+                    ) ?? turf.polygon([]);
+
+                const extentSource = insetMap.getSource('extent');
+                if (extentSource) {
+                    (extentSource as maplibregl.GeoJSONSource).setData(extent);
+                }
+
+                const shadowSource = insetMap.getSource('shadow');
+                if (shadowSource) {
+                    (shadowSource as maplibregl.GeoJSONSource).setData(shadow);
+                }
+            };
+
+            mainMap.on('move', syncInsetMap);
+            mainMap.on('zoom', syncInsetMap);
+        }
+    }, [mainMap, insetMap]);
+
+    return (
+        <Box
```

Comment:
Is it possible to add some margin? like it is completely stuck in the corner, it would look nice if it was a bit far from both the edges.
<img width="336" alt="Screenshot 2023-09-27 at 2 48 58 PM" src="https://github.com/Oceans-1876/challenger-portal/assets/32737711/bb95edf0-6f8f-4cd7-8bb4-8548a9e7da9c">

Reply:
Ahh I see. I will pull that and test it.